includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1546,19 +1540,36 @@ public function update_contact_lists( $email, $lists_to_add = [], $lists_to_remo
 		}
 		$mc = new Mailchimp( $this->api_key() );
 		try {
+			// Remove lists.
 			foreach ( $lists_to_remove as $list_id ) {
-				$list = $this->maybe_extract_group_or_tag_list( $list_id );
-				if ( $list ) {
-					$this->remove_group_or_tag_from_contact( $email, $list['id'], $list['list_id'], $list['type'] );
+				$list_obj = Subscription_List::from_public_id( $list_id );
+				if ( ! $list_obj ) {
 					continue;
 				}
```

Comment:
Should we worry (log) when a list object is not obtainable via its public ID?

Reply:
This is one of those things we talked about.. We are not very consistent how we treat this sort of error, in a list of lists to add, if one of them fails.

We should definitely improve it, but for now this does not really change the current behavior

(at some point I needed to draw a line where I would stop refactoring)